### PR TITLE
[rfxcom] Replaced usage of DatatypeConverter

### PR DESCRIPTION
**Pull request by martinvw**
Fixes: #3424

Lets wait for the build but then it should be fine :-)

### CODE REVIEW

**Comment by martinvw** [comment]
Pitty, we need the reverse as well, first time for an ESH PR :-(

File: addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/connector/RFXComTcpConnector.java [hunk 1]
```diff
@@ -88,7 +87,7 @@ public void disconnect() {
 
     @Override
     public void sendMessage(byte[] data) throws IOException {
-        logger.trace("Send data (len={}): {}", data.length, DatatypeConverter.printHexBinary(data));
+        logger.trace("Send data (len={}): {}", data.length, HexUtils.bytesToHex(data));
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/connector/RFXComTcpConnector.java]: The `HexUtils.bytesToHex()` method is applied every time. Should we encapsulate it by an `if (logger.isTraceEnabled())`?

File: addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/connector/RFXComSerialConnector.java [hunk 2]
```diff
@@ -120,7 +119,7 @@ public void sendMessage(byte[] data) throws IOException {
             throw new IOException("Not connected sending messages is not possible");
         }
 
-        logger.trace("Send data (len={}): {}", data.length, DatatypeConverter.printHexBinary(data));
+        logger.trace("Send data (len={}): {}", data.length, HexUtils.bytesToHex(data));
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/connector/RFXComSerialConnector.java]: The `HexUtils.bytesToHex()` method is applied every time. Should we encapsulate it by an `if (logger.isTraceEnabled())`?

File: addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/connector/RFXComJD2XXConnector.java [hunk 3]
```diff
@@ -101,7 +100,7 @@ public void disconnect() {
 
     @Override
     public void sendMessage(byte[] data) throws IOException {
-        logger.trace("Send data (len={}): {}", data.length, DatatypeConverter.printHexBinary(data));
+        logger.trace("Send data (len={}): {}", data.length, HexUtils.bytesToHex(data));
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/connector/RFXComJD2XXConnector.java]: The `HexUtils.bytesToHex()` method is applied every time. Should we encapsulate it by an `if (logger.isTraceEnabled())`?

File: addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComBridgeHandler.java [hunk 4]
```diff
@@ -300,10 +298,9 @@ public void packetReceived(byte[] packet) {
                             + "issue at the relevant tracker. Received message: {}", message);
                 }
             } catch (RFXComMessageNotImplementedException e) {
-                logger.debug("Message not supported, data: {}", DatatypeConverter.printHexBinary(packet));
+                logger.debug("Message not supported, data: {}", HexUtils.bytesToHex(packet));
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComBridgeHandler.java]: The `HexUtils.bytesToHex()` method is applied every time. Should we encapsulate it by an `if (logger.isDebugEnabled())`?

File: addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComBridgeHandler.java [hunk 5]
```diff
@@ -259,8 +258,7 @@ public void packetReceived(byte[] packet) {
                                 }
 
                                 if (setMode != null) {
-                                    logger.debug("Setting RFXCOM mode using: {}",
-                                            DatatypeConverter.printHexBinary(setMode));
+                                    logger.debug("Setting RFXCOM mode using: {}", HexUtils.bytesToHex(setMode));
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComBridgeHandler.java]: The `HexUtils.bytesToHex()` method is applied every time. Should we encapsulate it by an `if (logger.isDebugEnabled())`?

File: addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComBridgeHandler.java (hunk 4 above)
Comment (martinvw) [reply addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComBridgeHandler.java]: This one is an exceptional situation so less critical and the only line in the catch, I think the added / gained value her is minimal

**Comment by martinvw** [comment]
Thanks @cweitkamp I updated the PR :-)

**Comment by martinvw** [comment]
Eclipse reverted back part of my changes I will have to fix this later have to head to my work now :-)

**Comment by martinvw** [comment]
@wborn now its fine, I just messed up some imports and there was one last occurence of the `DataTypeConverter` with a newline in between...

**Pull request merged**